### PR TITLE
Miscellaneous changes

### LINKED ISSUE #1220: Double free in http server destruction
Noticed this for an unrelated thing. In http/server.hpp

```
  virtual ~HTTPServer()
  {
    LOG_STACK_TRACE_POINT;

    auto socketWeak = socket_;
    auto accepWeak  = acceptor_;

    networkManager_.Post([socketWeak, accepWeak] {
      auto            socket   = socketWeak.lock();
      auto            acceptor = accepWeak.lock();
      std::error_code dummy;
      if (socket)
      {
        socket->shutdown(asio::ip::tcp::socket::shutdown_both, dummy);
        socket->close(dummy);
      }

      if (acceptor)
      {
        acceptor->close(dummy);
      }
    });

    /* manager_.reset(); */
  }
```

The manager has a double free if you uncomment the line doing the reset.

### PR DESCRIPTION
**Pull request by wkaluza**
Closes #1220

### CODE REVIEW

File: libs/http/src/module.cpp [hunk 1]
```diff
@@ -17,18 +17,16 @@
...
 
 bool NormalAccessAuthentication(HTTPRequest const &req)
 {
-  if (req.authentication_level() < AuthenticationLevel::DEFAULT_LEVEL)
-  {
-    return false;
-  }
-
-  return true;
+  return !(req.authentication_level() < AuthenticationLevel::DEFAULT_LEVEL);
```
Comment (bipll) [inline libs/http/src/module.cpp]: Whis is the same as `return req.authentication_level() >= AuthenticationLevel::DEFAULT_LEVEL;`

File: libs/math/include/math/meta/math_type_traits.hpp [hunk 2]
```diff
@@ -72,9 +72,6 @@ using IfNoVectorSupportFor = std::enable_if_t<!HasVectorSupport<T>::value, R>;
 /// TYPES INDIRECTED FROM META / TYPE_TRAITS ///
 ////////////////////////////////////////////////
 
-template <typename DataType, typename ReturnType>
-using IfIsNotImplemented = fetch::meta::IfIsNotImplemented<DataType, ReturnType>;
-
 template <typename DataType>
 using IfIsUnsignedInteger = fetch::meta::IfIsUnsignedInteger<DataType>;
```
Comment (bipll) [inline libs/math/include/math/meta/math_type_traits.hpp]: How about `using fetch::meta::IfIsUnsignedInteger;`?

**Review by bipll — CHANGES_REQUESTED** [review]

File: libs/http/src/module.cpp (hunk 1 above)
Comment (wkaluza) [reply libs/http/src/module.cpp]: done

File: libs/math/include/math/meta/math_type_traits.hpp (hunk 2 above)
Comment (wkaluza) [reply libs/math/include/math/meta/math_type_traits.hpp]: :+1:

File: libs/meta/include/meta/value_util.hpp [hunk 3]
```diff
@@ -86,22 +86,5 @@ constexpr auto Accumulate(F &&f, A &&a, B &&b,
                     std::forward<Seq>(seq)...);
 }
 
-// Simple uses of Accumulate().
-// Since Accumulate() itself operates on packs rather than ranges,
-// there's no special overload to calculate simple sum, like in STL.
-template <typename H, typename... Ts>
-constexpr auto Sum(H &&h, Ts &&... ts) noexcept(
```
Comment (bipll) [inline libs/meta/include/meta/value_util.hpp:93]: These two were actually introduced as docs, to demonstrate how `Accumulate` can compensate for the lack of fold-expressions.

Comment (wkaluza) [reply libs/meta/include/meta/value_util.hpp:93]: It's dead code.

File: libs/http/include/http/server.hpp [hunk 4]
```diff
@@ -95,9 +95,6 @@ class HTTPServer : public AbstractHTTPServer
         acceptor->close(dummy);
       }
     });
-
-    // TODO (issue 1220): This appears to cause a double free due to a race
-    /* manager_.reset(); */
```
Comment (ejfitzgerald) [inline libs/http/include/http/server.hpp:100]: I believe this is an invalid setup issue can you mark this issue as closed with this PR

**Review by ejfitzgerald — APPROVED** [review]

**Pull request merged**